Sources/PluginLibrary/Array+Extensions.swift

```diff
@@ -0,0 +1,22 @@
+// Sources/PluginLibrary/Array+Extensions.swift - Additions to Arrays
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+extension Array {
+  /// Like map, but calls the transform with the index and value.
+  func enumeratedMap<T>(_ transform: (Int, Element) throws -> T) rethrows -> [T] {
```

Comment:
macOS app on my laptop -

```swift
extension Array {
  /// Like map, but calls the transform with the index and value.
  func enumeratedMap<T>(_ transform: (Int, Element) throws -> T) rethrows -> [T] {
    var i: Int = -1
    return try map {
      i += 1
      return try transform(i, $0)
    }
  }
}

struct Blah {
  let title: String
  let index: Int
  init(title: String, index: Int) {
    self.title = title
    self.index = index
  }
}

@NSApplicationMain
class AppDelegate: NSObject, NSApplicationDelegate {

  func applicationDidFinishLaunching(_ aNotification: Notification) {
    let l = ["abc", "def", "ghi", "jkl"]

    var start = DispatchTime.now()
    for _ in 0...5000000 {
      let _ = l.enumerated().map {
        return Blah(title: $0.element, index: $0.offset)
      }
    }
    var end = DispatchTime.now()
    var elapsed = end.uptimeNanoseconds - start.uptimeNanoseconds
    var intervalSeconds = Double(elapsed) / 1000000000
    print("Using enumerated: \(intervalSeconds)")

    start = DispatchTime.now()
    for _ in 0...5000000 {
      let _ = l.enumeratedMap {
        return Blah(title: $1, index: $0)
      }
    }
    end = DispatchTime.now()
    elapsed = end.uptimeNanoseconds - start.uptimeNanoseconds
    intervalSeconds = Double(elapsed) / 1000000000
    print("Using enumeratedMap: \(intervalSeconds)")
  }
}
```

Release build:
Using enumerated: 3.694987967
Using enumeratedMap: 0.961241992

Debug build:
Using enumerated: 20.038512905
Using enumeratedMap: 8.521299144

Want me to add that in a comment within the extension?